### PR TITLE
🌐 [i18n-KO] Translated `main_classes/callback.md` to Korean

### PR DESCRIPTION
**Pull request by Jwaminju**
# What does this PR do?

Translated the `main_classes/callback.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [ ] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

@junejae, @Jwaminju, @010kim, @4N3MONE, @boyunJang 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)
@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/main_classes/callback.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    def on_train_begin(self, args, state, control, **kwargs):
+        print("Starting training")
+
+
+trainer = Trainer(
+    model,
+    args,
+    train_dataset=train_dataset,
+    eval_dataset=eval_dataset,
+    callbacks=[MyCallback],  # We can either pass the callback class this way or an instance of it (MyCallback())
```
Comment (4N3MONE) [inline docs/source/ko/main_classes/callback.md:114]: 주석 부분도 번역이 필요할 것 같습니다!

File: docs/source/ko/main_classes/callback.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+)
+```
+
+또 다른 콜백을 등록하는 방법은 `trainer.add_callback()`을 호출하는 것입니다:
+
+```python
+trainer = Trainer(...)
+trainer.add_callback(MyCallback)
+# Alternatively, we can pass an instance of the callback class
```
Comment (4N3MONE) [inline docs/source/ko/main_classes/callback.md:123]: 주석 부분도 번역이 필요할 것 같습니다!(2)

Comment (Jwaminju) [reply docs/source/ko/main_classes/callback.md]: 감사합니다!

File: docs/source/ko/main_classes/callback.md (hunk 1 above)
Comment (Jwaminju) [reply docs/source/ko/main_classes/callback.md]: 오! 놓칠 수 있었던 부분인데, 감사합니다!

File: docs/source/ko/main_classes/callback.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 콜백 [[callbacks]]
+
+콜백은 PyTorch [`Trainer`]의 훈련 루프 동작을 사용자 정의할 수 있는 객체입니다
```
Comment (Jwaminju) [inline docs/source/ko/main_classes/callback.md:19]:
```suggestion
콜백은 PyTorch [`Trainer`]의 반복 훈련 동작을 사용자 정의할 수 있는 객체입니다
```

File: docs/source/ko/main_classes/callback.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+(이 기능은 TensorFlow에서는 아직 구현되지 않았습니다). 콜백은 훈련 루프의 상태를
```
Comment (Jwaminju) [inline docs/source/ko/main_classes/callback.md:20]:
```suggestion
(이 기능은 TensorFlow에서는 아직 구현되지 않았습니다). 콜백은 반복 훈련의 상태를
```

File: docs/source/ko/main_classes/callback.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+검사하여 진행 상황 보고, TensorBoard 또는 기타 머신 러닝 플랫폼에 로그 등의 작업을 할 수 있으며, 
+결정(예: 조기 종료)도 내릴 수 있습니다.
+
+콜백은 [`TrainerControl`] 객체를 반환하는 것 외에는 훈련 루프에서 어떤 것도 변경할 수 없는
```
Comment (Jwaminju) [inline docs/source/ko/main_classes/callback.md:24]:
```suggestion
콜백은 [`TrainerControl`] 객체를 반환하는 것 외에는 반복 훈련에서 어떤 것도 변경할 수 없는
```

File: docs/source/ko/main_classes/callback.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+"읽기 전용" 코드 조각입니다. 훈련 루프에 변경이 필요한 사용자 정의 작업이 필요한 경우,
```
Comment (Jwaminju) [inline docs/source/ko/main_classes/callback.md:25]:
```suggestion
"읽기 전용" 코드 조각입니다. 반복 훈련에 변경이 필요한 사용자 정의 작업이 필요한 경우, 
```

File: docs/source/ko/main_classes/callback.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+- [`~integrations.ClearMLCallback`]는 [clearml](https://github.com/allegroai/clearml)이 설치되어 있으면 사용됩니다.
+- [`~integrations.DagsHubCallback`]는 [dagshub](https://dagshub.com/)이 설치되어 있으면 사용됩니다.
+- [`~integrations.FlyteCallback`]는 [flyte](https://flyte.org/)가 설치되어 있으면 사용됩니다.
+- [`~integrations.DVCLiveCallback`]는 [dvclive](https://dvc.org/doc/dvclive)가 설치되어 있으면 사용됩니다.
+
+패키지가 설치되어 있지만 해당 통합을 사용하지 않으려면, `TrainingArguments.report_to`를 사용하고 싶은 통합 목록으로 변경할 수 있습니다 (예: `["azure_ml", "wandb"]`).
+
+콜백을 구현하는 주요 클래스는 [`TrainerCallback`]입니다. 이 클래스는 [`Trainer`]를 
+인스턴스화하는 데 사용된 [`TrainingArguments`]를 가져오고, 해당 Trainer의 내부 상태를 
+[`TrainerState`]를 통해 접근할 수 있으며, [`TrainerControl`]을 통해 훈련 루프에서 일부
```
Comment (Jwaminju) [inline docs/source/ko/main_classes/callback.md:53]:
```suggestion
[`TrainerState`]를 통해 접근할 수 있으며, [`TrainerControl`]을 통해 반복 훈련에서 일부 
```

File: docs/source/ko/main_classes/callback.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 콜백 [[callbacks]]
+
+콜백은 PyTorch [`Trainer`]의 훈련 루프 동작을 사용자 정의할 수 있는 객체입니다
+(이 기능은 TensorFlow에서는 아직 구현되지 않았습니다). 콜백은 훈련 루프의 상태를
+검사하여 진행 상황 보고, TensorBoard 또는 기타 머신 러닝 플랫폼에 로그 등의 작업을 할 수 있으며,
```
Comment (Jwaminju) [inline docs/source/ko/main_classes/callback.md:21]:
```suggestion
검사하여 (진행 상황 보고, TensorBoard 또는 기타 머신 러닝 플랫폼에 로그 남기기 등) 
```

File: docs/source/ko/main_classes/callback.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+rendered properly in your Markdown viewer.
+
+-->
+
+# 콜백 [[callbacks]]
+
+콜백은 PyTorch [`Trainer`]의 훈련 루프 동작을 사용자 정의할 수 있는 객체입니다
+(이 기능은 TensorFlow에서는 아직 구현되지 않았습니다). 콜백은 훈련 루프의 상태를
+검사하여 진행 상황 보고, TensorBoard 또는 기타 머신 러닝 플랫폼에 로그 등의 작업을 할 수 있으며, 
+결정(예: 조기 종료)도 내릴 수 있습니다.
```
Comment (Jwaminju) [inline docs/source/ko/main_classes/callback.md:22]:
```suggestion
결정(예: 조기 종료)을 내릴 수 있습니다.
```

File: docs/source/ko/main_classes/callback.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+콜백은 [`TrainerControl`] 객체를 반환하는 것 외에는 반복 훈련에서 어떤 것도 변경할 수 없는
+"읽기 전용" 코드 조각입니다. 반복 훈련에 변경이 필요한 사용자 정의 작업이 필요한 경우, 
+[`Trainer`]를 서브클래스로 만들어 필요한 메소드들을 오버라이드해야 합니다 (예시는 [trainer](trainer)를 참조하세요).
+
+기본적으로 `TrainingArguments.report_to`는 `"all"`로 설정되어 있으므로, [`Trainer`]는 다음 콜백을 사용합니다.
+
+- [`DefaultFlowCallback`]는 로그, 저장, 평가에 대한 기본 동작을 처리합니다.
+- [`PrinterCallback`] 또는 [`ProgressCallback`]는 진행 상황을 표시하고 로그를 출력합니다 
+  (첫 번째는 [`TrainingArguments`]를 통해 tqdm을 비활성화하면 사용되고, 그렇지 않으면 두 번째가 사용됩니다).
```
Comment (boyunJang) [inline docs/source/ko/main_classes/callback.md:32]:
```suggestion
  ([`TrainingArguments`]를 통해 tqdm을 비활성화하면 첫 번째 콜백이 사용되고, 그렇지 않으면 두 번째가 사용됩니다).
```
번역을 살짝 고쳐보았습니다!

File: docs/source/ko/main_classes/callback.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+- [`~integrations.AzureMLCallback`]는 [azureml-sdk](https://pypi.org/project/azureml-sdk/)가 설치되어
+ 있으면 사용됩니다.
+- [`~integrations.CodeCarbonCallback`]는 [codecarbon](https://pypi.org/project/codecarbon/)이 설치되어
+ 있으면 사용됩니다.
+- [`~integrations.ClearMLCallback`]는 [clearml](https://github.com/allegroai/clearml)이 설치되어 있으면 사용됩니다.
+- [`~integrations.DagsHubCallback`]는 [dagshub](https://dagshub.com/)이 설치되어 있으면 사용됩니다.
+- [`~integrations.FlyteCallback`]는 [flyte](https://flyte.org/)가 설치되어 있으면 사용됩니다.
+- [`~integrations.DVCLiveCallback`]는 [dvclive](https://dvc.org/doc/dvclive)가 설치되어 있으면 사용됩니다.
+
+패키지가 설치되어 있지만 해당 통합을 사용하지 않으려면, `TrainingArguments.report_to`를 사용하고 싶은 통합 목록으로 변경할 수 있습니다 (예: `["azure_ml", "wandb"]`).
```
Comment (boyunJang) [inline docs/source/ko/main_classes/callback.md:49]:
```suggestion
패키지가 설치되어 있지만 해당 통합 기능을 사용하고 싶지 않다면, `TrainingArguments.report_to`를 사용하고자 하는 통합 기능 목록으로 변경할 수 있습니다 (예: `["azure_ml", "wandb"]`).
```

File: docs/source/ko/main_classes/callback.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 콜백 [[callbacks]]
+
+콜백은 PyTorch [`Trainer`]의 반복 훈련 동작을 사용자 정의할 수 있는 객체입니다
+(이 기능은 TensorFlow에서는 아직 구현되지 않았습니다). 콜백은 반복 훈련의 상태를
```
Comment (boyunJang) [inline docs/source/ko/main_classes/callback.md:20]:
```suggestion
(이 기능은 TensorFlow에서는 아직 구현되지 않았습니다). 콜백은 반복 학습의 상태를
```
훈련 -> 학습으로 변경했습니다.

File: docs/source/ko/main_classes/callback.md [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 콜백 [[callbacks]]
+
+콜백은 PyTorch [`Trainer`]의 반복 훈련 동작을 사용자 정의할 수 있는 객체입니다
```
Comment (boyunJang) [inline docs/source/ko/main_classes/callback.md:19]:
```suggestion
콜백은 PyTorch [`Trainer`]의 반복 학습 동작을 사용자 정의할 수 있는 객체입니다
```
훈련 -> 학습으로 변경했습니다.

File: docs/source/ko/main_classes/callback.md [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+(이 기능은 TensorFlow에서는 아직 구현되지 않았습니다). 콜백은 반복 훈련의 상태를
+검사하여 (진행 상황 보고, TensorBoard 또는 기타 머신 러닝 플랫폼에 로그 남기기 등) 
+결정(예: 조기 종료)을 내릴 수 있습니다.
+
+콜백은 [`TrainerControl`] 객체를 반환하는 것 외에는 반복 훈련에서 어떤 것도 변경할 수 없는
```
Comment (boyunJang) [inline docs/source/ko/main_classes/callback.md:24]:
```suggestion
콜백은 [`TrainerControl`] 객체를 반환하는 것 외에는 반복 학습에서 어떤 것도 변경할 수 없는
```

File: docs/source/ko/main_classes/callback.md [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+"읽기 전용" 코드 조각입니다. 반복 훈련에 변경이 필요한 사용자 정의 작업이 필요한 경우,
```
Comment (boyunJang) [inline docs/source/ko/main_classes/callback.md:25]:
```suggestion
"읽기 전용" 코드 조각입니다. 반복 학습에 변경이 필요한 사용자 정의 작업이 필요한 경우, 
```

File: docs/source/ko/main_classes/callback.md [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+- [`~integrations.ClearMLCallback`]는 [clearml](https://github.com/allegroai/clearml)이 설치되어 있으면 사용됩니다.
+- [`~integrations.DagsHubCallback`]는 [dagshub](https://dagshub.com/)이 설치되어 있으면 사용됩니다.
+- [`~integrations.FlyteCallback`]는 [flyte](https://flyte.org/)가 설치되어 있으면 사용됩니다.
+- [`~integrations.DVCLiveCallback`]는 [dvclive](https://dvc.org/doc/dvclive)가 설치되어 있으면 사용됩니다.
+
+패키지가 설치되어 있지만 해당 통합을 사용하지 않으려면, `TrainingArguments.report_to`를 사용하고 싶은 통합 목록으로 변경할 수 있습니다 (예: `["azure_ml", "wandb"]`).
+
+콜백을 구현하는 주요 클래스는 [`TrainerCallback`]입니다. 이 클래스는 [`Trainer`]를 
+인스턴스화하는 데 사용된 [`TrainingArguments`]를 가져오고, 해당 Trainer의 내부 상태를 
+[`TrainerState`]를 통해 접근할 수 있으며, [`TrainerControl`]을 통해 반복 훈련에서 일부
```
Comment (boyunJang) [inline docs/source/ko/main_classes/callback.md:53]:
```suggestion
[`TrainerState`]를 통해 접근할 수 있으며, [`TrainerControl`]을 통해 반복 학습에서 일부 
```

**Review by boyunJang — COMMENTED** [review]
번역 작업하느라 수고하셨습니다! 리뷰를 늦게 해드려서 죄송해요🙏
문장 조금씩 다듬어봤는데 확인 부탁드립니다:)

**Comment by Jwaminju** [comment]
Hello, @stevhliu May you please review this PR?
Thanks in advacne!

**Review by junejae — APPROVED** [review]
LGTM! 섬세한 번역 감사드립니다!

**Comment by 4N3MONE** [comment]
리뷰 완료하였습니다. 특별하게 고칠 사항 없는 것 같습니다. 고생하셨습니다~

**Review by stevhliu — APPROVED** [review]
Awesome, thank you! Just need to fix the toctree 🤗

File: docs/source/ko/_toctree.yml [hunk 17]
```diff
@@ -274,8 +274,8 @@
       title: (번역중) Auto Classes
     - local: in_translation
       title: (번역중) Backbones
-    - local: in_translation
-      title: (번역중) Callbacks
+    - local: callback
```
Comment (stevhliu) [inline docs/source/ko/_toctree.yml:277]:
```suggestion
    - local: main_classes/callback
```

**Comment by Jwaminju** [comment]
> Awesome, thank you! Just need to fix the toctree 🤗 

Great catch! Thanks!

**Pull request merged**